### PR TITLE
[UD][OPS-12302] Adding name parsing to remove titles before matching

### PR DESCRIPTION
**Pull request by Ufuoma-Daniel**

### CODE REVIEW

File: app/views/identityverification/EnterYourDateOfBirthPage.scala.html [hunk 1]
```diff
@@ -56,7 +56,7 @@
                     classes = "govuk-fieldset__legend--xl",
                     isPageHeading = true)))
             )
-        ).withFormField(form("date")))
+        ).withDayMonthYearFormField(form("date")))
```
Comment (Ufuoma-Daniel) [inline app/views/identityverification/EnterYourDateOfBirthPage.scala.html:59]: .WithFormField was deprecated with the move to `play-frontend-hmrc-play-30" % "10.1.0"`

File: test/util/NameParsingUtilSpec.scala [hunk 2]
```diff
@@ -20,6 +20,48 @@ import testsupport.UnitSpec
...
+    result should be ("John Doe")
+  }
+
+  "removeTitleFromName should remove all titles from the name when multiple names are provided" in {
+    val result = NameParsingUtil.removeTitleFromName(titlesList, "Mr John Doe, Mrs. Jane Doe")
+    result should be ("John Doe, Jane Doe")
+  }
+
+  "removeTitleFromName should not remove parts of the name that match a title" in {
+    val result = NameParsingUtil.removeTitleFromName(titlesList, "John Doe")
```
Comment (barnesjake) [inline test/util/NameParsingUtilSpec.scala:61]: The test data (name) for this scenario doesn't look right to me, shouldn't it be something like `John Droe` or something similar?

File: test/util/NameParsingUtilSpec.scala [hunk 3]
```diff
@@ -20,6 +20,48 @@ import testsupport.UnitSpec
...
+    val result = NameParsingUtil.removeTitleFromName(titlesList, "Mr John Andrew Doe")
+    result should be ("John Andrew Doe")
+  }
+
+  "removeTitleFromName should remove the title even if it includes a fullstop" in {
+    val result = NameParsingUtil.removeTitleFromName(titlesList, "Mr. John Andrew Doe")
+    result should be ("John Andrew Doe")
+  }
+
+  "removeTitleFromName should remove the title from the name when the title is not at the start" in {
```
Comment (Ufuoma-Daniel) [inline test/util/NameParsingUtilSpec.scala:45]: I'm thinking about this scenario, and it's occurred to me that anyone that has a title as a real name would get filtered. e.g. Tim Miss, Rev Doe etc.

File: test/util/NameParsingUtilSpec.scala (hunk 2 above)
Comment (Ufuoma-Daniel) [reply test/util/NameParsingUtilSpec.scala:61]: Oh good spot, it's a leftover, that test is actually what the test on line 30 became:

```
"removeTitleFromName should return the same name when a titles letters is part of the name" in {
    val result = NameParsingUtil.removeTitleFromName(titlesList, "Mrian Revanish")
    result should be ("Mrian Revanish")
  }
```

File: test/util/NameParsingUtilSpec.scala (hunk 3 above)
Comment (Ufuoma-Daniel) [reply test/util/NameParsingUtilSpec.scala:45]: After some discussion, changes required.

**Review by barnesjake — APPROVED** [review]

**Pull request merged**